src/app/(proper_react)/redesign/(authenticated)/user/dashboard/fix/leaked-passwords/leakedPasswordsData.tsx

```diff
@@ -0,0 +1,153 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import type { ReactNode } from "react";
+import passwordIllustration from "../images/leaked-passwords.svg";
+import securityQuestionsIllustration from "../images/security-questions.svg";
+import { SubscriberBreach } from "../../../../../../../../utils/subscriberBreaches";
+import { GuidedExperienceBreaches } from "../../../../../../../functions/server/getUserBreaches";
+import { getL10n } from "../../../../../../../functions/server/l10n";
+
+export type LeakedPasswordsContent = {
+  summary: string;
+  description: ReactNode;
+  recommendations?: {
+    title: string;
+    steps: ReactNode;
+    subtitle?: string;
+  };
+};
+
+export type LeakedPasswordsTypes = "password" | "security-question";
+
+export type LeakedPassword = {
+  type: LeakedPasswordsTypes;
+  title: string;
+  illustration: string;
+  content: LeakedPasswordsContent;
+};
+
+function getLeakedPasswords({
+  dataType,
+  breaches,
+}: {
+  dataType: string;
+  breaches: GuidedExperienceBreaches;
+}) {
+  const l10n = getL10n();
+
+  const findFirstUnresolvedBreach = (breachClassType: LeakedPasswordsTypes) => {
+    const leakedPasswordType =
+      breachClassType === "password" ? "passwords" : "securityQuestions";
+    return Object.values(breaches.passwordBreaches[leakedPasswordType]).find(
+      (breach) => !breach.isResolved
+    );
+  };
+
+  const unresolvedPasswordBreach = findFirstUnresolvedBreach("password");
+  const unresolvedSecurityQuestionsBreach =
+    findFirstUnresolvedBreach("security-question");
+
+  const getBreachInfo = (breach?: SubscriberBreach) => ({
+    name: breach ? breach.name : "",
+    breachDate: breach ? breach.breachDate : "",
+    breachSite: breach ? new URL(`https://${breach.domain}`) : "",
```

Comment:
Blocking:  We created a blocklist `HIBP_BREACH_DOMAIN_BLOCKLIST` of domains that changed ownership or are not available anymore. On the current site we do not just link to domains from breaches and this was the result of discussions and a conscious decision. We should show the external links only conditionally like we are doing here: https://github.com/mozilla/blurts-server/blob/main/src/utils/breachResolution.js#L119.

Reply:
Currently we are just not adding the link for these.

Reply:
Currently we just not add the link for these.